File: 35.Search_Insert_Position/search-insert-position.js
// Source : https://leetcode.com/problems/search-insert-position/
// Author : Kyon Huang
// Date   : 2017-09-17

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var searchInsert = function(nums, target) {
    var start = 0,
        end = nums.length - 1;
    
    while(start <= end) {
        let mid = (start + end) / 2;
        if(nums[mid] < target) {
             start = mid + 1;
        } 
        else if(nums[mid] > target){
            end = mid - 1;
        }
        else
            return mid;
    }

    return start
};

File: code/35.Search_Insert_Position/search-insert-position.java
// Source : https://leetcode.com/problems/search-insert-position/
// Author : Kyon Huang
// Date   : 2017-09-17

class Solution {
    public int searchInsert(int[] nums, int target) {
        int low = 0, high = nums.length-1;
        while(low <= high){
            int mid = (low+high) / 2;
            if(nums[mid] == target) return mid;
            else if(nums[mid] > target) high = mid-1;
            else low = mid+1;
        }
        return low;
    }
}